org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java

```diff
@@ -12369,9 +12364,7 @@ private void findTypesFromExpectedTypes(char[] token, Scope scope, ObjectVector
 				} else if (expectedType instanceof BaseTypeBinding) {
 					BaseTypeBinding baseType = (BaseTypeBinding) expectedType;
 					// only go further if we are completing on an array.
-					if (dimensions > 0 && proposeType &&
-							(!this.assistNodeIsConstructor || !allowingLongComputationProposals)
```

Comment:
`(!this.assistNodeIsConstructor || !allowingLongComputationProposals)` is gone now, could you explain why?

Reply:
mainly we don't need the allowingLongComputationProposals here, this was copied from `findNestedTypes` in the original commit which added support for array completions. Even though it make sense checking for this in nested types, the same doesn't make sense here.